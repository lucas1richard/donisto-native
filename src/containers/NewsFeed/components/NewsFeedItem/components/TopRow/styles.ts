import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  imgPlaceholder: {
    height: 32,
    width: 32,
    backgroundColor: '#f0f0f0',
    borderRadius: 16,
    marginRight: 12
  },
  nameWrapper: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  topWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 20,
    height: 58
  }
});

export default styles;
