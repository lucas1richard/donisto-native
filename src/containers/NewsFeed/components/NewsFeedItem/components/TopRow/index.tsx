import * as React from 'react';
import { View } from 'react-native';
import Txt from 'components/Txt';
import Touchable from 'components/Touchable';
import { Entypo } from '@expo/vector-icons';
import styles from './styles';

export interface TopRowProps {
  name: string;
}

class TopRow extends React.Component<TopRowProps> {
  render() {
    const { name } = this.props;
    return (
      <View style={styles.topWrapper}>
        <View style={styles.nameWrapper}>
          <View style={styles.imgPlaceholder} />
          <View>
            <Txt style={{ fontSize: 16 }}>
              {name}
            </Txt>
            <Txt style={{ fontSize: 11, color: 'rgb(171, 169,167)' }}>
              @xyz123
            </Txt>
          </View>
        </View>
        <Touchable>
          <View>
            <Entypo
              name="dots-three-horizontal"
              color="#d0d0d0"
              size={19}
            />
          </View>
        </Touchable>
      </View>
    );
  }
}

export default TopRow;