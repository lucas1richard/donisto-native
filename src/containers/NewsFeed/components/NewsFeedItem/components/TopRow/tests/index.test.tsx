import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TopRow, { TopRowProps } from '..';

describe('<TopRow />', () => {
  const renderComponent = (props: TopRowProps) => toJson(shallow(<TopRow {...props} />));
  it('renders', () => {
    expect(renderComponent({
      name: 'mockName'
    })).toMatchSnapshot();
  });
});