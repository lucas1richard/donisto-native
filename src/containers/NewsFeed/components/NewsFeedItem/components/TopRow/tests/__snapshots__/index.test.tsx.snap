// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopRow /> renders 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 58,
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "#f0f0f0",
          "borderRadius": 16,
          "height": 32,
          "marginRight": 12,
          "width": 32,
        }
      }
    />
    <Component>
      <Txt
        color="default"
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        mockName
      </Txt>
      <Txt
        color="default"
        style={
          Object {
            "color": "rgb(171, 169,167)",
            "fontSize": 11,
          }
        }
      >
        @xyz123
      </Txt>
    </Component>
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <Component>
      <Icon
        allowFontScaling={false}
        color="#d0d0d0"
        name="dots-three-horizontal"
        size={19}
      />
    </Component>
  </TouchableOpacity>
</Component>
`;
